features/frameworks/greentea-client/source/test_env.cpp

```diff
@@ -61,8 +61,10 @@ static void greentea_notify_version();
  * Rawserial object used to provide direct, raw serial communications
  * between the target and the host.
  */
-RawSerial greentea_serial(USBTX, USBRX); 
-
+RawSerial& greentea_serial() { 
+    static RawSerial serial(USBTX, USBRX);
```

Comment:
Is this guaranteed by the C++ standard to be initialized on first use or could it also be initialized at boot? Also, I doubt this is thread safe for GCC, although if utest calls this in its initialization sequence then it will be initialized in a single threaded manner. Maybe leave a comment here? Either that or you could ensure that it is thread safe by using the singleton_lock().

Also, @bridadan I know you made this change locally. Do you have any feedback on this? This version might be cleaner that using a singleton class.